janitor/functions.py

```diff
@@ -911,17 +918,29 @@ def rename_columns(df: pd.DataFrame, new_column_names: Dict) -> pd.DataFrame:
         import janitor
         df = pd.DataFrame(...).rename_columns({"old_column_name": "new_column_name"})
 
-    This is just syntactic sugar/a convenience function for renaming one column
-    at a time. If you are convinced that there are multiple columns in need of
-    changing, then use the :py:meth:`pandas.DataFrame.rename` method.
+    This is just syntactic sugar/a convenience function for renaming multiple columns
+    at a time. If you need to rename single column,
+    then use the `rename_column` method.
+
+    One of the new_column_names or function are a required parameter
 
     :param df: The pandas DataFrame object.
     :param new_column_names: A dictionary of old and new column names.
+    :param function: A function which should be applied to all the columns
     :returns: A pandas DataFrame with renamed columns.
+    :raises ValueError: if both new_column_names and function are None
     """  # noqa: E501
-    check_column(df, list(new_column_names.keys()))
 
-    return df.rename(columns=new_column_names)
+    if new_column_names is None and function is None:
+        raise ValueError(
+            "One of new_column_names or function must be provided"
+        )
+
+    if new_column_names is not None:
+        check_column(df, list(new_column_names.keys()))
```

Comment:
``check_column(df, new_column_names)`` works fine and is a bit cleaner, since by default , the keys of the dict are iterated.